Sources/AmazonLocationiOSAuthSDK/Client/AmazonLocationClient.swift

```diff
@@ -0,0 +1,119 @@
+            let resolver:AWSClientRuntime.StaticAWSCredentialIdentityResolver? =  try StaticAWSCredentialIdentityResolver(AWSCredentialIdentity(accessKey: credentials.accessKeyId, secret: credentials.secretKey, expiration: credentials.expiration, sessionToken: credentials.sessionToken))
+            
+            let clientConfig = try await LocationClient.LocationClientConfiguration(awsCredentialIdentityResolver: resolver, region: locationProvider.getRegion(), signingRegion: locationProvider.getRegion())
+            self.locationClient = LocationClient(config: clientConfig)
+        }
+    }
+    
+    public func sendRequest<T: Decodable, E: AmazonBaseErrorResponse & Error>(
+        serviceName: AmazonService,
+        endpoint: AmazonLocationEndpoint,
+        httpMethod: HTTPMethod,
+        requestBody: EncodableRequest?,
+        successType: T.Type,
+        errorType: E.Type
+    ) async throws -> AmazonLocationResponse<T, E> {
+        let url = URL(string: endpoint.url())!
+        var request = URLRequest(url: url)
+        request.httpMethod = httpMethod.rawValue
+        
+        // Encode request body if provided
+        if let requestBody = requestBody {
+            request.httpBody = try encodeRequestBody(requestBody)
+        }
+        
+        // Set default headers
+        var headers = HTTPHeaders()
+        headers.add(name: "Content-Type", value: "application/json")
+        
+        // Sign headers with AWS credentials or throw error if not available
+        if locationProvider.getCognitoProvider() != nil {
+            signHeaders(&request, url: url, serviceName: serviceName, httpMethod: httpMethod, headers: headers, requestData: request.httpBody)
+        } else if !endpoint.isApiKeyEndpoint() {
+            throw LocationProviderError.noCognitoOrApiKeyFound
+        }
+
+        // Execute the request and handle the response
+        return try await executeRequest(request, successType: successType, errorType: errorType)
+    }
+
+    private func encodeRequestBody(_ requestBody: EncodableRequest) throws -> Data {
+        do {
+            return try requestBody.toData()
+        } catch {
+            print("Error: Unable to encode request body as JSON")
+            throw error
+        }
+    }
+
+    private func signHeaders(_ request: inout URLRequest, url: URL, serviceName: AmazonService, httpMethod: HTTPMethod, headers: HTTPHeaders, requestData: Data?) {
+        
+        let cognitoCredentials = locationProvider.getCognitoProvider()!.getCognitoCredentials()
+        let signer = AWSSigner(credentials: cognitoCredentials!, serviceName: serviceName.rawValue, region: locationProvider.getRegion()!)
+        let signedHeaders = signer.signHeaders(url: url, method: httpMethod, headers: headers, body: requestData != nil ? .data(requestData!) : nil)
+        
+        for (name, value) in signedHeaders.allHeaders() {
+            request.addValue(value, forHTTPHeaderField: name)
+        }
+    }
```

Comment:
General question, is there not an open-source AWS SDK for swift that has a signer and other such helpers?

Reply:
All the AWS service will use the internal signer for cognito identity pool id. This signer is for external services such as MapLibre's map signing

Reply:
Thanks for the context!